# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1338325886 -7200
#      Tue May 29 23:11:26 2012 +0200
# Branch gui
# Node ID 06abf71d9083b043340ce5045bdf810ccf814236
# Parent  7623bece76df9e48296c38fb99c645424d341005
Fixed bug with current directory.

* MainWindow: Removed if condition to check whether a path is already in the QComboBox's model.

diff --git a/gui/src/MainWindow.cpp b/gui/src/MainWindow.cpp
--- a/gui/src/MainWindow.cpp
+++ b/gui/src/MainWindow.cpp
@@ -146,26 +146,23 @@ void
 MainWindow::resetWindows ()
 {
   // TODO: Implement.
 }
 
 void
 MainWindow::updateCurrentWorkingDirectory (QString directory)
 {
-  if(m_currentDirectoryComboBox->findText (directory) < 0)
+  if (m_currentDirectoryComboBox->count () > 31)
     {
-      if (m_currentDirectoryComboBox->count () > 31)
-        {
-          m_currentDirectoryComboBox->removeItem (0);
-        }
-      m_currentDirectoryComboBox->addItem (directory);
-      int index = m_currentDirectoryComboBox->findText (directory);
-      m_currentDirectoryComboBox->setCurrentIndex (index);
+      m_currentDirectoryComboBox->removeItem (0);
     }
+  m_currentDirectoryComboBox->addItem (directory);
+  int index = m_currentDirectoryComboBox->findText (directory);
+  m_currentDirectoryComboBox->setCurrentIndex (index);
 }
 
 void
 MainWindow::changeCurrentWorkingDirectory ()
 {
   QString selectedDirectory =
       QFileDialog::getExistingDirectory(this, tr ("Set working direcotry"));
 
